python/idsse_common/idsse/common/vectaster.py

```diff
@@ -196,59 +237,161 @@ def geographic_polygon_to_pixel(
 
     Args:
-        line_string (LineString): Shapely geometry with vertices defined by x,y
-        dist (float, optional): Distance cutoff used to determine in a pixel is "near".
-                                Defaults to 0.25.
+        linestring (LineString): Shapely geometry with vertices defined by x,y in pixel space
 
     Returns:
         Tuple[numpy.array]: First array represent the x-coordinate and the seconde the y.
     """
-    xmin, ymin, xmax, ymax = line_string.bounds
-    xmin = int(floor(xmin))
-    xmax = int(ceil(xmax))
-    ymin = int(floor(ymin))
-    ymax = int(ceil(ymax))
-
-    points = []
-    for y_coord in range(ymin, ymax):
-        for x_coord in range(xmin, xmax):
-            points.append(Point(x_coord, y_coord))
-
-    points_near_line_string = [(pnt.x, pnt.y) for pnt in points if line_string.distance(pnt) < dist]
-    return tuple(numpy.array(dim_coord) for dim_coord in map(list, zip(*points_near_line_string)))
+    return _make_numpy(_pixels_for_linestring(linestring))
 
 
 def pixels_in_polygon(poly: Polygon) -> Tuple[numpy.ndarray]:
-    """Determine the pixels that are within the specified Polygon
+    """Determine the pixels that represent the specified Polygon
 
     Args:
-        poly (Polygon): Shapely geometry with vertices defined by x,y
+        poly (Polygon): Shapely geometry with vertices defined by x,y in pixel space
 
     Returns:
         Tuple[numpy.array]: First array represent the x-coordinate and the seconde the y.
     """
-    xmin, ymin, xmax, ymax = poly.bounds
-    xmin = int(floor(xmin))
-    xmax = int(ceil(xmax))
-    ymin = int(floor(ymin))
-    ymax = int(ceil(ymax))
-
-    # create prepared polygon
-    # prep_poly = prep(poly)
-
-    # construct a rectangular mesh
-    points = []
-    for y_coord in range(ymin, ymax):
-        for x_coord in range(xmin, xmax):
-            points.append((x_coord, y_coord))
-
-    # validate if each point falls inside shape using
-    # the prepared polygon
-    # points_in_poly = list(filter(prep_poly.contains, points))
-    points_in_poly = [pnt for pnt in points if contains_xy(poly, *pnt)]
-    return tuple(numpy.array(dim_coord) for dim_coord in map(list, zip(*points_in_poly)))
+    pixels = _pixels_for_polygon(poly.exterior)
+    for inner in poly.interiors:
+        pixels_on_inner_edge = _pixels_for_linestring(inner)
+        for pixel in _pixels_for_polygon(inner):
+            if pixel not in pixels_on_inner_edge:
+                pixels.remove(pixel)
+
+    return _make_numpy(pixels)
+
+
+def _make_numpy(points: List[Pixel]) -> Tuple[numpy.ndarray]:
+    """Map a list of (x,y) tuples to a tuple of x values (as numpy array) and y values (numpy array)
+
+    Args:
+        points (List[Pixel]): list of (x,y) coordinates
+
+    Returns:
+        Tuple[numpy.ndarray]: Same coordinates but restructured
+    """
+    return tuple(numpy.array(dim_coord, dtype=numpy.int64) for dim_coord in map(list, zip(*points)))
```

Comment:
I think I used something very similar in GridProj to do the same kind of thing: transform a list of tuples (x, y) into two numpy arrays (x_coordinates, y_coordinates):

https://github.com/NOAA-GSL/idss-engine-commons/blob/main/python/idsse_common/idsse/common/grid_proj.py#L299-303

Could this one here be a public function so both this and GridProj could use it?

Reply:
I think making this a shared function would be a good idea, but I wouldn't have the function in vectaster.py since this module imports from grid_proj.py and then it would import from vectaster.py. Maybe just move it utils.py or some other base utility module.